custom_components/grocy/binary_sensor.py

```diff
@@ -1,45 +1,136 @@
 
 
-class GrocyBinarySensor(GrocyEntity, BinarySensorEntity):
-    """Grocy binary_sensor class."""
+@dataclass
+class GrocyBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Grocy binary sensor entity description."""
+
+    attributes_fn: Callable[[List[Any]], Mapping[str, Any] | None] = lambda _: None
+    exists_fn: Callable[[List[str]], bool] = lambda _: True
+    entity_registry_enabled_default: bool = False
+
+
+BINARY_SENSORS: tuple[GrocyBinarySensorEntityDescription, ...] = (
+    GrocyBinarySensorEntityDescription(
+        key=ATTR_EXPIRED_PRODUCTS,
+        name="Grocy expired products",
+        icon="mdi:delete-alert-outline",
+        exists_fn=lambda entities: ATTR_EXPIRED_PRODUCTS in entities,
+        attributes_fn=lambda data: {
+            "expired_products": [x.as_dict() for x in data],
+            "count": len(data),
+        },
+    ),
+    GrocyBinarySensorEntityDescription(
+        key=ATTR_EXPIRING_PRODUCTS,
+        name="Grocy expiring products",
+        icon="mdi:clock-fast",
+        exists_fn=lambda entities: ATTR_EXPIRING_PRODUCTS in entities,
+        attributes_fn=lambda data: {
+            "expiring_products": [x.as_dict() for x in data],
+            "count": len(data),
+        },
+    ),
+    GrocyBinarySensorEntityDescription(
+        key=ATTR_MISSING_PRODUCTS,
+        name="Grocy missing products",
+        icon="mdi:flask-round-bottom-empty-outline",
+        exists_fn=lambda entities: ATTR_MISSING_PRODUCTS in entities,
+        attributes_fn=lambda data: {
+            "missing_products": [x.as_dict() for x in data],
+            "count": len(data),
+        },
+    ),
+    GrocyBinarySensorEntityDescription(
+        key=ATTR_OVERDUE_CHORES,
+        name="Grocy overdue chores",
+        icon="mdi:alert-circle-check-outline",
+        exists_fn=lambda entities: ATTR_OVERDUE_CHORES in entities,
+        attributes_fn=lambda data: {
+            "overdue_chores": [x.as_dict() for x in data],
+            "count": len(data),
+        },
+    ),
+    GrocyBinarySensorEntityDescription(
+        key=ATTR_OVERDUE_TASKS,
+        name="Grocy overdue tasks",
+        icon="mdi:alert-circle-check-outline",
+        exists_fn=lambda entities: ATTR_OVERDUE_TASKS in entities,
+        attributes_fn=lambda data: {
+            "overdue_tasks": [x.as_dict() for x in data],
+            "count": len(data),
+        },
+    ),
+)
+
+
+class GrocyBinarySensorEntity(GrocyEntity, BinarySensorEntity):
+    """Grocy binary sensor entity definition."""
+
+    def __init__(
+        self,
+        coordinator: GrocyDataUpdateCoordinator,
+        description: BinarySensorEntityDescription,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize binary sensor entity."""
+        super().__init__(coordinator)
+        self._attr_name = description.name
+        self._attr_unique_id = f"{config_entry.entry_id}{description.key.lower()}"
+        self.entity_description = description
```

Comment:
This is the same for the sensor as well (other than typing for the description), so this should move to `GrocyEntity`